src/Compilers/Core/Portable/Operations/IRecursivePatternOperation.cs

```diff
@@ -13,6 +13,25 @@ namespace Microsoft.CodeAnalysis.Operations
     /// </remarks>
     public interface IRecursivePatternOperation : IPatternOperation
     {
+        /// <summary>
+        /// The (explicit or implicit) type accepted for the recursive pattern.
+        /// </summary>
+        ITypeSymbol MatchedType { get; }
+        /// <summary>
+        /// The Deconstruct symbol, if any, used for the deconstruction subpatterns.
+        /// </summary>
+        ISymbol DeconstructSymbol { get; }
+        /// <summary>
+        /// If there is a deconstruction or positional subpattern, this contains the patterns contained within it.
+        /// If there is no deconstruction subpattern, this is a default immutable array.
+        /// </summary>
+        ImmutableArray<IPatternOperation> DeconstructionSubpatterns { get; }
+        /// <summary>
+        /// If there is a property subpattern, this contains the
+        /// <see cref="ISymbol"/>/<see cref="IPatternOperation"/> pairs within it.
+        /// If there is no property subpattern, this is a default immutable array.
+        /// </summary>
+        ImmutableArray<(ISymbol, IPatternOperation)> PropertySubpatterns { get; }
```

Comment:
>(ISymbol, IPatternOperation) [](start = 23, length = 28)

Need to get some feedback on whether we should be using Tuples here. We could potentially replace with a struct instead.